src/metabase/util/schema.clj

```diff
@@ -364,8 +375,14 @@
 (def ParameterSourceOptions
   "Schema for valid source_options within a Parameter"
   ;; TODO: This should be tighter
-  {(s/optional-key :values) (s/cond-pre [s/Any])
-   (s/optional-key :card_id) IntGreaterThanZero})
+  {;; for source_type = 'static-list'
+   (s/optional-key :values)          (s/cond-pre [s/Any])
+
+   ;; for source_type = 'card'
+   (s/optional-key :card_id)         IntGreaterThanZero
+   (s/optional-key :value_field_ref) FieldRef
+   ;; label_field_ref is optional
+   (s/optional-key :label_field_ref) FieldRef})
```

Comment:
yea it should. but I'm having a branch depend on this so lemme have this merge and the rename branch merge. Then I'll do that. otherwise it'll be conflict hell :'(